walkeeper/src/send_wal.rs

```diff
@@ -75,9 +92,16 @@ impl postgres_backend::Handler for SendWalHandler {
         } else if query_string.starts_with(b"START_REPLICATION") {
             ReplicationConn::new(pgb).run(self, pgb, &query_string)?;
         } else if query_string.starts_with(b"START_WAL_PUSH") {
-            ReceiveWalConn::new(pgb)
-                .run(self)
-                .with_context(|| "failed to run ReceiveWalConn")?;
+            // this solution with pageserver connstr maybe non obvious, maybe it makes more sense to have it in START_WAL_PUSH directly
```

Comment:
Is there anything against providing pageserver_connstr in START_WAL_PUSH query? I looked through the code, it seems that it would be more convenient to use something like `START_WAL_PUSH {pageserver_connstr}` and will not require some hacks like checking for "pageserver" in application_name.

Reply:
There's also an option to pass connstr in ProposerGreeting, but it may be less convenient due to serialization of variable length string.

Reply:
Agree. I've changed it. This is indeed looks cleaner.